eng/build.ps1

```diff
@@ -267,6 +267,10 @@ foreach ($argument in $PSBoundParameters.Keys)
   }
 }
 
+if ($env:TreatWarningsAsErrors -eq 'false') {
+  $arguments += " -warnAsError 0"
+}
```

Comment:
I think it would make sense to put this and the corresponding shell script change into the eng/common/build.ps1 and eng/common/build.sh files so that all repos benefit from it.

Reply:
Should we test drive it here first though?

Reply:
Sure, let's merge this in now. Still, it would be good to follow-up and make the corresponding change in Arcade and then revert the changes in our build.ps1 and build.sh scripts.